wgpu-hal/src/gles/egl.rs

```diff
@@ -618,8 +619,13 @@ impl crate::Instance<super::Api> for Instance {
                 .unwrap();
             (display, Some(Arc::new(library)), WindowKind::AngleX11)
         } else {
-            log::info!("Using default platform");
-            let display = egl.get_display(egl::DEFAULT_DISPLAY).unwrap();
+            log::info!("No windowing system present. Using surfaceless platform");
+            let egl = egl
+                .upcast::<egl::EGL1_5>()
```

Comment:
note that surfaceless is also available in EGL1_4, according to https://www.khronos.org/registry/EGL/extensions/MESA/EGL_MESA_platform_surfaceless.txt

Reply:
`getPlatformDisplay` is EGL 1.5 feature, so I did the upcast.
https://www.khronos.org/registry/EGL/sdk/docs/man/html/eglGetPlatformDisplay.xhtml

Btw happy New Year! Best wishes!

Reply:
This path should also support EGL1_4 with https://www.khronos.org/registry/EGL/extensions/EXT/EGL_EXT_platform_base.txt present, which provides the entry point.

Reply:
Hmm, not sure. Examples I found for surfaceless used getPlatformDisplay, as well as other platforms out there in Wgpu's EGL code

Reply:
We are using `get_platform_display` in the backend if it's available. But here for your PR we are modifying the very last fallback path. It has to be minimally restrictive. Hence why it's important to support EGL1_4 with EXT_platform_base extension. If you are unsure how to get it going, we can leave that for a follow-up.